pkg/server/plugin/nodeattestor/aws/iid.go

```diff
@@ -194,85 +193,76 @@ func (p *IIDAttestorPlugin) Attest(stream nodeattestorv0.NodeAttestor_AttestServ
 	}
 
 	if shouldCheckBlockDevice {
-		err = p.checkBlockDevice(instance)
-		if err != nil {
-			return iidError.New("failed aws ec2 attestation: %w", err)
+		if err = p.checkBlockDevice(instance); err != nil {
+			return status.Errorf(codes.Internal, "failed aws ec2 attestation: %v", err)
 		}
 	}
 
-	agentID, err := makeSpiffeID(c.trustDomain, c.pathTemplate, validDoc, tags)
+	agentID, err := makeSpiffeID(c.trustDomain, c.pathTemplate, attestationData, tags)
 	if err != nil {
-		return iidError.New("failed to create spiffe ID: %w", err)
+		return status.Errorf(codes.Internal, "failed to create spiffe ID: %v", err)
 	}
 
 	attested, err := p.IsAttested(stream.Context(), agentID.String())
 	switch {
 	case err != nil:
 		return err
 	case attested:
-		return iidError.New("IID has already been used to attest an agent")
+		return status.Error(codes.InvalidArgument, "IID has already been used to attest an agent")
```

Comment:
Hmm, maybe permission denied in this case?
```suggestion
		return status.Error(codes.PermissionDenied, "IID has already been used to attest an agent")
```